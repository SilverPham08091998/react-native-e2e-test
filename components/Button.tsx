import { Pressable, PressableProps, StyleProp, TextStyle, ViewStyle } from 'react-native';
import Layout from 'constants/Layout';
import Colors from 'constants/Colors';
import { Text, TextProps } from './Text';
import { ThemeProps, useThemeColor } from './Themed';

const base: StyleProp<ViewStyle> = {
  flexDirection: 'row',
  paddingHorizontal: Layout.spacing.lg,
  height: Layout.button,
  justifyContent: 'center',
  alignItems: 'center',
  borderRadius: Layout.button,
};

const presets = {
  primary: base as StyleProp<ViewStyle>,
  secondary: [base, { backgroundColor: Colors.light.tabIconDefault }] as StyleProp<ViewStyle>,
  disabled: [base, { backgroundColor: Colors.light.disabled }] as StyleProp<ViewStyle>,
};
type Presets = keyof typeof presets;
export type ButtonProps = PressableProps &
  ThemeProps & {
    title: string;
    preset?: Presets;
    titleProps?: TextProps;
  };
export function Button(props: ButtonProps) {
  const {
    title,
    preset = 'primary',
    style,
    darkColor,
    lightColor,
    titleProps,
    ...pressableProps
  } = props;
  const isDisabled = pressableProps.disabled || preset === 'disabled';
  const btnPreset = isDisabled ? 'disabled' : preset;

  const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, btnPreset);
  const buttonStyle = [presets[preset], { backgroundColor }, style] as PressableProps['style'];

  const color = useThemeColor(
    { light: titleProps?.lightColor, dark: titleProps?.darkColor },
    isDisabled ? 'textBtnDisabled' : 'textBtn',
  );
  const textStyle = [{ color }] as TextStyle;
  return (
    <Pressable
      accessible
      accessibilityRole="button"
      accessibilityLabel={title}
      accessibilityHint={title}
      {...pressableProps}
      style={buttonStyle}
    >
      <Text preset="primary" style={textStyle}>
        {title}
      </Text>
    </Pressable>
  );
}
