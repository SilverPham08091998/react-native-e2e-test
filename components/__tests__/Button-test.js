import * as React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '../Button';

it(`renders correctly`, () => {
  const onPress = jest.fn()
  const tree = renderer.create(<Button onPress={onPress} title="Primary" preset="primary" />).toJSON();

  expect(tree).toMatchSnapshot();
});
