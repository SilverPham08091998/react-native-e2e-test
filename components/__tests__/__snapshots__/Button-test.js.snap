// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityHint="Primary"
  accessibilityLabel="Primary"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 50,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 16,
      },
      {
        "backgroundColor": "#33449b",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 17,
          "fontWeight": "500",
          "letterSpacing": 0.3,
          "lineHeight": 20,
        },
        {
          "color": "#000",
        },
        [
          {
            "color": "#fff",
          },
        ],
      ]
    }
  >
    Primary
  </Text>
</View>
`;
